crates/oxc_regular_expression/src/display.rs

```diff
@@ -0,0 +1,524 @@
+        CharacterKind::UnicodeEscape => {
+            // we remove the leading `0x` of our 4 digit hex number.
+            let hex = &format!("{cp:#4X}")[2..];
+            if hex.len() <= 4 {
+                format!(r"\u{hex}")
+            } else {
+                format!(r"\u{{{hex}}}")
+            }
+        }
+        CharacterKind::HexadecimalEscape => {
+            // we remove the leading `0x` of our 2 digit hex number.
+            let hex = &format!("{cp:#2X}")[2..];
+            format!(r"\x{hex}")
+        }
+        CharacterKind::Octal => {
+            let octal = format!("{cp:o}");
+            format!(r"\{octal}")
+        }
+        CharacterKind::SingleEscape => match ch {
+            '\n' => String::from(r"\n"),
+            '\r' => String::from(r"\r"),
+            '\t' => String::from(r"\t"),
+            '\u{b}' => String::from(r"\v"),
+            '\u{c}' => String::from(r"\f"),
+            '\u{8}' => String::from(r"\b"),
+            '\u{2D}' => String::from(r"\-"),
+            _ => format!(r"\{ch}"),
+        },
+    };
+
+    (result, peeked)
+}
+
+#[cfg(test)]
+mod test {
+    use oxc_allocator::Allocator;
+    type Case<'a> = (
+        &'a str,
+        /* expected display */ Option<&'a str>,
+        /* test with unicode mode as well */ bool,
+    );
+
+    static CASES: &[Case] = &[
+        ("/ab/", None, true),
+        ("/abc/i", None, true),
+        ("/a*?/i", None, true),
+        ("/emo👈🏻ji/u", None, true),
+        ("/ab|c/i", None, true),
+        ("/a|b+|c/i", None, true),
+        ("/(?=a)|(?<=b)|(?!c)|(?<!d)/i", None, true),
+        (r"/(cg)(?<n>cg)(?:g)/", None, true),
+        (r"/^(?=ab)\b(?!cd)(?<=ef)\B(?<!gh)$/", None, true),
+        (r"/^(?<!ab)$/", None, true),
+        (r"/[abc]/", None, true),
+        (r"/[a&&b]/v", None, true),
+        (r"/[a--b]/v", None, true),
+        (r"/[^a--b--c]/v", None, true),
+        (r"/[a[b[c[d[e[f[g[h[i[j[k[l]]]]]]]]]]]]/v", None, true),
+        (r"/[\q{abc|d|e|}]/v", None, true),
+        (r"/\p{Basic_Emoji}/v", None, true),
+        (r"/[|\]]/", None, true),
+        (r"/c\]/", None, true),
+        ("/a{0}|b{1,2}|c{3,}/i", None, true),
+        (r"/Em🥹j/", None, true),
+        (r"/\n\cM\0\x41\./", None, true),
+        (r"/\n\cM\0\x41\u1234\./u", None, true),
+        (r"/[\bb]/", None, true),
+        (r"/\d+/gu", None, true),
+        (r"/\D/gu", None, true),
+        (r"/\w/gu", None, true),
+        (r"/\w+/gu", None, true),
+        (r"/\s/g", None, true),
+        (r"/\s+/g", None, true),
+        (r"/\t\n\v\f\r/", None, true),
+        // we lose the flags ordering
+        ("/abcd/igv", Some("/abcd/giv"), true),
+        // we lose the flags ordering
+        (r"/\d/ug", Some(r"/\d/gu"), true),
+        // NOTE: we capitalize all hex values.
+        (r"/\n\cM\0\x41\u{1f600}\./u", Some(r"/\n\cM\0\x41\u{1F600}\./u"), true),
+        // TODO: tests below do **NOT** parse with unicode mode on
+        (r"/c]/", None, false),
```

Comment:
These cases where the last tuple item is `false` fail to parse with Unicode mode. Is it a known issue? Something I've done wrong or should we create an issue for it?

Reply:
`/c]/` is invalid with unicode mode.

![image](https://github.com/user-attachments/assets/484c2097-25ca-4f37-91f1-d56e159757c2)

Reply:
Oh, I used https://regexr.com/ to test it. It doesn't show this syntax error, It's a bug on their side I guess.
I'll remove that todo comment since it is the intended behavior(not compiling with unicode flag)

Reply:
![image](https://github.com/user-attachments/assets/c95ee05d-2449-476d-8454-76f1b05f5523) 👀

Reply:
That error is always there, doesn't have anything to do with the `c]` pattern.

Reply:
Ops, I've mistaken it for the Unicode warning.